streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/ClientState.java

```diff
@@ -59,29 +60,31 @@
 
     private int capacity;
 
-    public ClientState() {
-        this(0);
+    public ClientState(final Map<String, String> clientTags) {
+        this(0, clientTags);
     }
 
-    ClientState(final int capacity) {
+    ClientState(final int capacity, final Map<String, String> clientTags) {
```

Comment:
I'm wondering could we keep the original constructor and pass empty map into the new one? So that we don't have to make changes to the old caller. That is:

```
public ClientState() {
    this(0, Collections.emptyMap());
}

ClientState(final int capacity) {
    this(capacity, Collections.emptyMap());
}
```
WDYT?

Reply:
Thanks for the feedback. No objections from my side. The reason why I avoided that, was because to make sure that client tags are always passed, to emphasise that it's mandatory parameter when constructing the `ClientState` object. Please note that we have made `ClientState#clientTags` immutable. So there're no setters for the client tags.
But if you feel like it's better to default to empty map, happy to change it.
Will wait for your response on this.

Reply:
This is an internal class, so I think it won't be changed/used many times. I think change to my previous suggestion is better. Thanks.